enterprise/backend/test/metabase_enterprise/serialization/v2/extract_test.clj

```diff
@@ -488,6 +489,65 @@
+                                                             :personal_owner_id mark-id}]
+                       Collection [{coll2-eid :entity_id}   {:name "Some Collection"}]
+                       Database   [{db-id        :id}       {:name "My Database"}]
+                       Table      [{no-schema-id :id}       {:name "Schemaless Table" :db_id db-id}]
+                       Field      [{measure-field-id :id}   {:name "Field for measure" :table_id no-schema-id}]
+                       Field      [{dimension-field-id :id} {:name "Field for dimension" :table_id no-schema-id}]
+                       Card       [{card-id  :id
+                                    card-eid :entity_id}    {:creator_id  mark-id}]
+                       Newmetric  [{metric-id  :id
+                                    metric-eid :entity_id}  {:creator_id    mark-id
+                                                             :collection_id coll-id
+                                                             :card_id       card-id
+                                                             :measure       [:sum [:field measure-field-id nil]]
+                                                             :dimensions    ["created_at"
+                                                                             [:field dimension-field-id {:temporal-unit :year}]]}]]
+      (testing "vanilla user-created newmetric"
+        (let [ser (serdes.base/extract-one "Newmetric" {} (select-one "Newmetric" [:= :id metric-id]))]
+          (is (schema= {:serdes/meta   (s/eq [{:model "Newmetric" :id metric-eid}])
+                        :creator_id    (s/eq "mark@direstrai.ts")
+                        :collection_id (s/eq coll-eid)
+                        :measure       (s/eq ["sum" [:field ["My Database" nil "Schemaless Table" "Field for measure"] nil]],)
+                        :dimensions    (s/eq ["created_at" [:field ["My Database" nil "Schemaless Table" "Field for dimension"] {:temporal-unit :year}]])
+                        :created_at    OffsetDateTime
+                        :updated_at    OffsetDateTime
+                        s/Keyword      s/Any}
+                       ser))
+          (is (not (contains? ser :id)))
+
+          (testing "newmetric depend on their Card and Collection, and also anything referenced in the dimension and measure"
+            (is (= #{[{:model "Database" :id "My Database"}
+                      {:model "Table" :id "Schemaless Table"}
+                      {:model "Field" :id "Field for dimension"}]
+                     [{:model "Database" :id "My Database"}
+                      {:model "Table" :id "Schemaless Table"}
+                      {:model "Field" :id "Field for measure"}]
+                     [{:model "Card" :id card-eid}]
+                     [{:model "Collection" :id coll-eid}]}
+                   (set (serdes.base/serdes-dependencies ser)))))))
+
+      (testing "collection filtering based on :user option"
```

Comment:
This test looks like it was borrowed from the tests for `Collection`s. If the goal is to test the `extract-query` on `Newmetric`, this should be checking that with `:user` set, a `Newmetric` in a `:user`-owned collection is extracted but one in another collection is not.

Reply:
Isn't it:
- if `user` is set, returns things that are not a personal collection and thing in the collection of `user`?
- if `user` is not set, returns things that are not in a personal collections

Reply:
I updated the tests!

Reply:
Ah, yes. My description was a bit too glib. You've implemented it correctly.